internal/database/memory/table.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package memory
+
+import (
```

Comment:
They are similar for all methods but `NewWriteBatch()`. The problem is the writeBatch implementation for both packages is quite different, so I don't think this can be abstracted away really. Unless we would use our own in-memory write batch for both database implementations, but I would rather let it flexible + use the badger write batch. Feel free to give it a stab in a draft PR if you want, but I couldn't make it work